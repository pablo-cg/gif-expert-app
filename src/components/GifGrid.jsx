import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifCard } from './GifCard';
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <>
      <h3>{isLoading ? 'Cargando...' : category}</h3>
      {!isLoading && (
        <div className="card-grid">
          {images.map((image) => (
            <GifCard key={image.id} {...image} />
          ))}
        </div>
      )}
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
