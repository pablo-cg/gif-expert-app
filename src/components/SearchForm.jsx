import { useState } from 'react';
import PropTypes from 'prop-types';

export const SearchForm = ({ onSearchCategory }) => {
  const [inputSearch, setInputSearch] = useState('');

  const search = (event) => {
    event.preventDefault();

    if (!inputSearch.trim()) return;

    onSearchCategory(inputSearch.trim());
    setInputSearch('');
  };

  return (
    <form aria-label="form" onSubmit={search}>
      <input
        id="input-search"
        type="text"
        placeholder="Buscar"
        value={inputSearch}
        onChange={({ target }) => setInputSearch(target.value)}
      />
      <small className="input-helper">Presiona enter para buscar</small>
    </form>
  );
};

SearchForm.propTypes = {
  onSearchCategory: PropTypes.func.isRequired,
};
