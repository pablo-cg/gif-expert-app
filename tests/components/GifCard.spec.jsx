import { render, screen } from '@testing-library/react';
import { GifCard } from '../../src/components/GifCard';

const title = 'Un título';
const url = 'https://localhost/algo.jpg';

describe('<GifCard /> Tests', () => {
  test('Debe hacer match con el snapshot', () => {
    const { container } = render(<GifCard title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test('Debe mostar la imagen con el src y alt de los props', () => {
    render(<GifCard title={title} url={url} />);

    const { src, alt } = screen.getByRole('img');

    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('Debe mostrar el título', () => {
    render(<GifCard title={title} url={url} />);

    expect(screen.getByText(title)).toBeTruthy();
  });
});
