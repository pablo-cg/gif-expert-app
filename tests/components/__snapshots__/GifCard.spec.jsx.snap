// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifCard /> Tests Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Un título"
      src="https://localhost/algo.jpg"
    />
    <p>
      Un título
    </p>
  </div>
</div>
`;
